tests/unit/test_cmac.cpp

```diff
@@ -0,0 +1,244 @@
+/*
```

Comment:
Should we also test that invoking `update()` multiple times works as expected?

Reply:
I would not expect any issues here, but sure, why not.
I will also note that on the list of potentiall improvements for the HMAC implementation.

Reply:
Done.